lib/govuk_component/button_helpers.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module GovukComponent
+  module ButtonHelpers
+    def govuk_button(content = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button', tag_options)
+      tag_options[:data] = { module: 'govuk-button', 'prevent-double-click': 'true' }
+      is_disabled(tag_options)
+
+      if block_given?
+        tag.button(tag_options, &block)
+      else
+        tag.button(content, tag_options)
+      end
+    end
+
+    def govuk_button_secondary(content = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button--secondary', tag_options)
+
+      govuk_button(content, tag_options, &block)
+    end
+
+    def govuk_button_warning(content = nil, **tag_options, &block)
+      tag_options = prepend_classes('govuk-button--warning', tag_options)
+
+      govuk_button(content, tag_options, &block)
+    end
+
+    def govuk_button_start(name = nil, options = nil, **tag_options, &block)
+      options = name if block_given?
+      url = url_for(options)
+      tag_options[:href] ||= url
+      tag_options = prepend_classes('govuk-button--start', tag_options)
+      govuk_link_button_options(tag_options)
+      content = block_given? ? capture(&block) : name || url
```

Comment:
Does this break the functionality? Just trying to `a.` reduce number of lines for this method and `b.` ensure that if an href is provideded directly as an option it becomes the fallback at line 35 - I think currently only an href in the options would be the `options` would be the fallback?

```suggestion
      tag_options[:href] ||= url_for(options)
       ...
      content = block_given? ? capture(&block) : name || tag_options[:href]
```

Reply:
happy to pair on this as i reckon there should be a few more tests around calling with various options too, to make it more reusable.

Reply:
as discussed this follows current rails helper `link_to` logic (where the actual URL and the url displayed can be different